mathesar/api/exceptions/exceptions.py

```diff
@@ -44,3 +56,331 @@ class GenericAPIError(MathesarAPIException):
     def __init__(self, error_body_list, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR):
         self.detail = [error_body._asdict() for error_body in error_body_list]
         self.status_code = status_code
+
+
+class UniqueViolationAPIException(MathesarAPIException):
+    def __init__(
+            self,
+            exception,
+            error_code=ErrorCodes.UniqueViolation.value,
+            message="This column has non-unique values so a unique constraint cannot be set",
+            field=None,
+            details=None,
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
+    ):
+        exception_detail = get_default_exception_detail(exception, error_code, message, field, details)._asdict()
+        self.detail = [exception_detail]
+        self.status_code = status_code
+
+
+class NotFoundAPIException(MathesarAPIException):
+    def __init__(
+            self,
+            exception,
+            error_code=ErrorCodes.NotFound.value,
+            message=None,
+            field=None,
```

Comment:
Do we need to specify arguments to exception classes when they're `None`? Can't you use `**kwargs` and handle `message`, `field`, and `details` in the `__init__` method of the `MathesarAPIException` class? That will reduce boilerplate code for new exceptions.

Reply:
It is nice to see what arguments are accepted by a method, rather than having to check the docs or inspect the method. I moved certain fields which don't have to be overridden outside of the `init` method.